ansible/roles/operator-pipeline/templates/openshift/tasks/index-signing-image-check.yml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: index-signing-image-check
+spec:
+  description: |
+    A task to check whether bootstrapped index image can be signed.
+    Only images that haven't been published as GA can be signed.
+  params:
+    - name: pipeline_image
+      description: A docker image of operator-pipeline-images for the steps to run in.
+    - name: image_pullspec
+      description: A index image pullspec to check
+    - name: registry_auth_secret_name
+      default: "index-img-signing-pipeline-registry-auth"
+    - name: pyxis_env
+      description: Environment to run in. Can be one of [dev, qa, stage, prod]
+      default: prod
+  results:
+    - name: docker_references
+      description: Comma-separated list of indices
+    - name: manifest_digests
+      description: Comma-separated list of manifest digests retrieved from registry
+
+  volumes:
+    - name: registry-auth
+      secret:
+        secretName: "$(params.registry_auth_secret_name)"
+
+  steps:
+    - name: check-image-in-pyxis
+      image: "$(params.pipeline_image)"
+      script: |
+        #! /usr/bin/env bash
+        set -xe
+
+        PYXIS_ENV=$(params.pyxis_env)
+        PYXIS_BASE_URL="https://catalog.redhat.com/api/containers"
+        if [ "$PYXIS_ENV" != "prod" ]; then
+          PYXIS_BASE_URL="https://catalog.stage.redhat.com/api/containers"
+        fi
+
+        FULL_PYXIS_URL="$PYXIS_BASE_URL/v1/operators/indices?filter=path==$(params.image_pullspec)"
+
+        # Use squid proxy to access non prod environments
+        curl -x http://squid.corp.redhat.com:3128 -s -X GET $FULL_PYXIS_URL > /tmp/pyxis_response.json
```

Comment:
Even before calling the pyxis api, shouldn't we check that the pullspec (excluding the tag) points to a valid index? Otherwise the pyxis call would return nothing and therefore the pipeline would go ahead with the signing.

Reply:
That's one thing I thought I added but forgot. The task now contains a check for allowed repositories.